include/aspect/mesh_deformation/diffusion.h

```diff
@@ -0,0 +1,185 @@
+        Diffusion();
+
+        /**
+         * Initialize function, which sets the start time and
+         * start timestep of diffusion.
+         */
+        void initialize() override;
+
+        /**
+         * The update function sets the current time and determines
+         * whether diffusion should be applied in this timestep.
+         */
+        void update() override;
+
+        /**
+         * A function that creates constraints for the velocity of certain mesh
+         * vertices (e.g. the surface vertices) for a specific boundary.
+         * The calling class will respect
+         * these constraints when computing the new vertex positions.
+         */
+        void
+        compute_velocity_constraints_on_boundary(const DoFHandler<dim> &mesh_deformation_dof_handler,
+                                                 ConstraintMatrix &mesh_velocity_constraints,
+                                                 const std::set<types::boundary_id> &boundary_id) const override;
+
+        /**
+         * Declare parameters for the diffusion of the surface.
+         */
+        static
+        void declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Parse parameters for the diffusion of the surface.
+         */
+        void parse_parameters (ParameterHandler &prm) override;
+
+      private:
+        /**
+         * Compute the surface velocity from a difference
+         * in surface height given by the solution of
+         * the hillslope diffusion problem.
+         */
+        void diffuse_boundary (const DoFHandler<dim> &free_surface_dof_handler,
+                               const IndexSet &mesh_locally_owned,
+                               const IndexSet &mesh_locally_relevant,
+                               LinearAlgebra::Vector &output,
+                               const std::set<types::boundary_id> boundary_id) const;
+
+        /**
+         * Check that the size of the next time step is not larger than the conduction
+         * timestep based on the mesh size and the diffusivity on each cell along the
+         * surface. The computed time step has to satisfy the CFL
+         * number chosen in the input parameter file on each cell of the mesh.
+         */
+        void check_diffusion_time_step (const DoFHandler<dim> &mesh_deformation_dof_handler,
+                                        const std::set<types::boundary_id> &boundary_ids) const;
+
+        /**
+         * The hillslope transport coefficient or diffusivity [m2/s]
+         * used in the hillslope diffusion of the deformed
+         * surface.
+         */
+        double diffusivity;
+
+        /**
+         * The start time of the model run
+         * used to determine whether diffusion should
+         * be applied.
+         */
+        double start_time;
+
+        /**
+         * The current time used to determine whether diffusion should
+         * be applied.
+         */
+        double current_time;
```

Comment:
Is this and the previous value equal to what the `Simulator` class stores? I would very much prefer if we could query it from there, rather than duplicating information -- this kind of thing has a tendency to go out of sync, with very difficult-to-find bugs.